chrome/model.js

```diff
@@ -223,6 +219,79 @@ GmailToTrello.Model.prototype.loadTrelloMembers = function(boardId) {
     });
 };
 
+GmailToTrello.Model.prototype.loadThreadTrelloCards = function () {
+    var self = this;
+
+    self.isThreadCardsLoaded = true;
+
+    // get cards associated with the thread on the current thread
+    // i would love to do this with the message ids, but there's no way to get all of them to search
+    // so let's use the subject and time method
+    if (!self.gmail || !self.gmail.subject || self.gmail.subject.length < 1) {
+        gtt_log('loadThreadTrelloCards: Current page is not a thread');
+        self.trello.threadCards = [];
+        self.event.fire('onLoadThreadTrelloCardsSuccess');
+    }
+    else {
+        gtt_log('loadThreadTrelloCards: Cards for current thread');
+        var query = encodeURIComponent(self.gmail.subject);
+        Trello.get('search', {'query': query, 'modelTypes': 'cards'}, function(data) {
+
+            if (!data || !data.hasOwnProperty('cards')) {
+                data = {'cards': []}
+            }
+
+            // need to make sure the time matches
+            // because i may not be in the same time zone as when it was set, give a 24-hour window on either side
+            var cards = [];
+            data['cards'].forEach(function (card) {
+                var addCard = false;
+                var matches = card.desc.match(/&within=1d&date=(\S+)/g);
+                if (matches) {
+                  matches.forEach(function (str) {
+                      var cardTime = Date.parse(decodeURIComponent(str.substring(16)).replace(' at ', ' ')).getTime();
+                      self.gmail.messageTimes.forEach(function (messageTime) {
+                          var diff = cardTime - messageTime;
+                          if (diff < 86400000 && diff > -86400000) {
+                              addCard = true;
+                          }
+                      });
+                  });
+                }
+                if (addCard) {
+                    cards.push(card);
+                }
+            });
+
+            self.trello.threadCards = cards;
+            self.sortThreadCards();
+            self.event.fire('onLoadThreadTrelloCardsSuccess');
+
+        }, function failure(data) {
+            self.event.fire('onAPIFailure', {data:data});
+        });
+    }
+
+};
+
+GmailToTrello.Model.prototype.sortThreadCards = function () {
+
+    var self = this;
+    self.trello.threadCards.sort(function (a,b) {
+        if (!a.closed && b.closed) { return -1; }
+        if (a.closed && !b.closed) { return 1; }
+        if (!a.dueComplete && b.dueComplete) { return -1; }
+        if (a.dueComplete && !b.dueComplete) { return 1; }
+        if (a.due && !b.due) { return -1; }
+        if (b.due && !a.due) { return 1; }
+        if (a.due && b.due) {
+            return new Date(a.due.replace('T',' ')).getTime() - new Date(b.due.replace('T',' ')).getTime();
+        }
+        return 0;
+    });
```

Comment:
This is used before displaying the threadCards to sort them in the most reasonable way. Soonest due cards show up first, then later due cards, then complete cards, then archived cards.